tools/chplcheck/src/rules.py

```diff
@@ -17,93 +17,136 @@
 # limitations under the License.
 #
 
+import re
+import typing
+
 import chapel
 from chapel import *
-import re
+from driver import LintDriver
+from fixits import ChapelFixit, range_to_text
+from rule_types import FullBasicRuleResult, FullAdvancedRuleResult
+
 
-def name_for_linting(node):
+def name_for_linting(context: Context, node: NamedDecl) -> str:
     name = node.name()
 
     # Strip dollar signs.
     name = name.replace("$", "")
 
+    # if the name is inside an internal module, strip leading `_` and `chpl_`
+    if context.is_bundled_path(node.location().path()):
+        if name.startswith("_"):
+            name = name[1:]
+        if name.startswith("chpl_"):
+            name = name[5:]
```

Comment:
I specifically removed this prior, I think. The reason being that `chpl_` isn't tread specially always -- only when `chpl_` is passed in as a prefix. I think the principled way to make this work is to read from that "internal prefix" array that we get from argparse.

Reply:
I agree, however we don't have a good way to thread that from `LintDriver` to the rules. For now, I have removed this and replaced with a todo. I will also open an issue on it

Reply:
Opened https://github.com/chapel-lang/chapel/issues/24812